Parse/src/main/java/com/parse/ParseFileController.java

```diff
@@ -125,44 +132,53 @@ public void clearCache() {
     if (cancellationToken != null && cancellationToken.isCancelled()) {
       return Task.cancelled();
     }
-    final File file = getCacheFile(state);
+    final File cacheFile = getCacheFile(state);
     return Task.call(new Callable<Boolean>() {
       @Override
       public Boolean call() throws Exception {
-        return file.exists();
+        return cacheFile.exists();
       }
     }, Task.BACKGROUND_EXECUTOR).continueWithTask(new Continuation<Boolean, Task<File>>() {
       @Override
       public Task<File> then(Task<Boolean> task) throws Exception {
         boolean result = task.getResult();
         if (result) {
-          return Task.forResult(file);
+          return Task.forResult(cacheFile);
         }
         if (cancellationToken != null && cancellationToken.isCancelled()) {
           return Task.cancelled();
         }
 
+        // Generate the temp file path for caching ParseFile content based on ParseFile's url
+        // The reason we do not write to the cacheFile directly is because there is no way we can
+        // verify if a cacheFile is complete or not. If download is interrupted in the middle, next
+        // time when we download the ParseFile, since cacheFile has already existed, we will return
+        // this incomplete cacheFile
```

Comment:
It is useful but we have already had cacheFile. Before we fetch, we check whether the cacheFile exist or not. So instead of checking file exists or not in disk, you want to keep a in-memory map to make the checking faster?